src/spring-cloud/azext_spring_cloud/application_configuration_service.py

```diff
@@ -0,0 +1,223 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=unused-argument, logging-format-interpolation, protected-access, wrong-import-order, too-many-lines
+import json
+
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core.util import sdk_no_wait
+from knack.log import get_logger
+from knack.util import CLIError
+from msrestazure.tools import resource_id
+
+from .vendored_sdks.appplatform.v2022_01_01_preview import models
+
+APPLICATION_CONFIGURATION_SERVICE_NAME = "applicationConfigurationService"
+RESOURCE_ID = "resourceId"
+
+RESOURCE_TYPE = "configurationServices"
+DEFAULT_NAME = "default"
+
+logger = get_logger(__name__)
+
+
+def application_configuration_service_show(cmd, client, service, resource_group):
+    return client.configuration_services.get(resource_group, service, DEFAULT_NAME)
+
+
+def application_configuration_service_clear(cmd, client, service, resource_group):
+    properties = models.ConfigurationServiceGitProperty()
+    acs_resource = models.ConfigurationServiceResource(properties=properties)
+    return client.configuration_services.begin_create_or_update(resource_group, service, DEFAULT_NAME, acs_resource)
+
+
+def application_configuration_service_git_add(cmd, client, service, resource_group,
+                                              name, patterns, uri, label,
+                                              search_paths=None,
+                                              username=None,
+                                              password=None,
+                                              host_key=None,
+                                              host_key_algorithm=None,
+                                              private_key=None,
+                                              strict_host_key_checking=None,
+                                              no_wait=False):
+    repo = models.ConfigurationServiceGitRepository(name=name, patterns=patterns, uri=uri, label=label)
+    repo = _replace_repo_with_input(repo, patterns, uri, label, search_paths, username, password, host_key, host_key_algorithm, private_key, strict_host_key_checking)
+
+    acs_resource = _get_or_default_acs_resource(client, resource_group, service)
+    repos = acs_resource.properties.settings.git_property.repositories
+    if next((r for r in repos if r.name == name), None) is not None:
+        raise CLIError("Repo '{}' already exists.".format(name))

```

Comment:
Please use specific error type from `azure.cli.core.azclierror` instead of `CLIError`.

Reply:
Updated, thanks!